pkg/provider/cloud/vsphere/provider.go

```diff
@@ -174,7 +100,8 @@ func (v *Provider) InitializeCloudProvider(ctx context.Context, cluster *kuberma
 
 	tagCategoryName := cluster.Spec.Cloud.VSphere.TagCategoryName
 	// We only need to fetch/create tag categories only if the user explicitly decides to use on.
-	if tagCategoryName != "" {
+	if force || tagCategoryName != "" {
```

Comment:
Same problem here, the finalizer is always added. In addition, `createTagCategory` does not check if the tag category already exists, so I would assume it fails on subsequent reconcile runs (or if the field is pre-set).

Reply:
Why? Here this doesn't happen. If we find a category id that exists already, we actually return from the function without adding any finalizers.

Reply:
True, I actually misunderstood what this does. My bad, should be fine.